etc/make_small_lib.pl

```diff
@@ -0,0 +1,335 @@
+#!/usr/bin/env perl
+
+use strict;
+use warnings;
+
+use Getopt::Long;
+use File::Basename 'basename';
+use File::Glob 'bsd_glob';
+use File::Spec;
+
+my $sep = File::Spec->catfile('', '');
+my %fast_functions = (
+       s_mp_balance_mul            => 1,
+       s_mp_exptmod_fast           => 1,
+       s_mp_invmod_fast            => 1,
+       s_mp_karatsuba_mul          => 1,
+       s_mp_karatsuba_sqr          => 1,
+       s_mp_montgomery_reduce_fast => 1,
+       s_mp_mul_digs_fast          => 1,
+       s_mp_mul_high_digs_fast     => 1,
+       s_mp_sqr_fast               => 1,
+       s_mp_toom_mul               => 1,
+       s_mp_toom_sqr               => 1
+   );
+my @dependency_list = ();
```

Comment:
this is for weak dependencies I assume?

Reply:
(Note to self: be a bit more verbose in commenting the code)

Yes.
But it is hardcoded and you can only switch *all* on/off for now.

It also ignores the extra-small alternatives to some functions as listed in the example for RSA in `tommath_superclass.h`